core/lib/components/balena/sdk.js

```diff
@@ -546,4 +547,79 @@ module.exports = class BalenaSDK {
 		supervisor = supervisor.replace(`",`, ``);
 		return supervisor
 	}
+
+	/** Downloads provided version of balenaOS using balenaSDK
+	 * @param The semver compatible balenaOS version that will be downloaded, example: `2.80.3+rev1.dev`. Default value: `latest` where latest development variant of balenaOS will be downloaded.
+	 * @param The device type for which balenaOS needs to be downloaded
+	*/
+	async fetchOS(version = "latest", deviceType) {
+		if (version === "latest") {
+			const versions = await this.balena.models.os.getSupportedVersions(deviceType);
+			// make sure we always flash the development variant
+			version = versions.latest.replace('prod', 'dev');
+		}
+
+		if (!fs.existsSync(`/data/images/`)) fs.mkdirSync(`/data/images/`);
```

Comment:
Is there a way we can make this not 'hard coded'? Maybe there is some way to use the config in `core/config/default.js`, although I'm not sure if that config is visible here

Reply:
I don't like this either, the long way of figuring out a persistent downloads/assets directory for Leviathan would be exactly how we form and specify the tmpdir. This way we are making the directory transparent and folks can check it out while it being always there. I can get it done with some more work, atm I left it hard-coded. Will plot a course for that one.

Reply:
We can make an issue and solve this in the future

Reply:
Yeah, let me just do that. I am actively working on it so looking forward to either dump this or uncomment it soon.